test/unit/lib/plugins/aws/lib/naming.test.js

```diff
@@ -973,4 +973,14 @@ describe('#naming()', () => {
       ).to.equal('custom/FnJoinRefApiGatewayRestApiexecuteapi');
     });
   });
+
+  describe('#getEcrRepositoryName', () => {
+    it('should correctly trim trailing dash and ensure no consecutive dashes are present', () => {
+      serverless.service.serviceObject = { name: 'service--with-weird-dashes---' };
```

Comment:
I'm not happy about that test, but I felt like trying to use `runServerless` to call `naming` methods might be a bit of an overkill - please let me know if you see a better approach to testing these methods :bow:

Reply:
I think good solution for that is addressing: https://github.com/serverless/serverless/issues/7056

Having that in, we do not have to handle this cases  as we can safely assume that format is valid.

Still, at this point let's keep what you proposed, I would just add a `TODO:` comment that it _replace_ logic should be removed with next major (assuming https://github.com/serverless/serverless/issues/7056 was addressed)

Reply:
Oh that would be ideal, however at this point I think that even with addressing #7056 - we have to introduce it in a non-breaking way, which means the normalization logic will have to be introduced at least in v2. :( I've added a `TODO` comment as you suggested to remove that logic with next major if #7056 is resolved.